test/e2e/testsuites/specs.go

```diff
@@ -310,3 +323,65 @@ func CreateVolumeSnapshotClass(client restclientset.Interface, namespace *v1.Nam
 
 	return tvsc, tvsc.Cleanup
 }
+func VerifySuccessfulReplicaAzVolumeAttachments(pod PodDetails, azDiskClient *azDiskClientSet.Clientset, storageClassParameters map[string]string, client clientset.Interface, namespace *v1.Namespace) (bool, *v1alpha1.AzVolumeAttachmentList, error) {
+	if storageClassParameters["maxShares"] == "" {
+		return true, nil, nil
+	}
+
+	var expectedNumberOfReplicas int
+	maxShares, err := strconv.ParseInt(storageClassParameters["maxShares"], 10, 0)
+	framework.ExpectNoError(err)
+	nodes := ListAzDriverNodeNames(azDiskClient.DiskV1alpha1().AzDriverNodes(azureconstants.AzureDiskCrdNamespace))
+	nodesAvailableForReplicas := len(nodes) - 1
+	if nodesAvailableForReplicas >= int(maxShares-1) {
+		expectedNumberOfReplicas = int(maxShares - 1)
+	} else {
+		expectedNumberOfReplicas = nodesAvailableForReplicas
+	}
+
+	for _, volume := range pod.Volumes {
+		if volume.PersistentVolume != nil {
+			replicaAttachments, err := GetReplicaAttachments(volume.PersistentVolume, client, namespace, azDiskClient)
+			framework.ExpectNoError(err)
+			numReplicaAttachments := len(replicaAttachments.Items)
+
+			if numReplicaAttachments != expectedNumberOfReplicas {
+				e2elog.Logf("expected %d replica attachments, found %d", expectedNumberOfReplicas, numReplicaAttachments)
+				return false, nil, nil
+			}
+
+			failedReplicaAttachments := v1alpha1.AzVolumeAttachmentList{}
+
+			for _, replica := range replicaAttachments.Items {
+				if replica.Status.State != "Attached" {
+					e2elog.Logf(fmt.Sprintf("found replica attachment %s, currently not attached", replica.Name))
+					failedReplicaAttachments.Items = append(failedReplicaAttachments.Items, replica)
+				} else {
+					e2elog.Logf(fmt.Sprintf("found replica attachment %s in attached state", replica.Name))
+
+				}
+			}
+			if len(failedReplicaAttachments.Items) > 0 {
+				return false, &failedReplicaAttachments, nil
+			}
+		}
+	}
+	return true, nil, nil
+}
+
+func GetReplicaAttachments(persistentVolume *v1.PersistentVolume, client clientset.Interface, namespace *v1.Namespace, azDiskClient *azDiskClientSet.Clientset) (*v1alpha1.AzVolumeAttachmentList, error) {
+	pv, err := client.CoreV1().PersistentVolumes().Get(context.TODO(), persistentVolume.Name, metav1.GetOptions{})
+	if err != nil {
+		ginkgo.Fail("failed to get persistent volume")
+	}
+	diskname, err := azureutils.GetDiskName(pv.Spec.CSI.VolumeHandle)
+	if err != nil {
+		ginkgo.Fail("failed to get persistent volume diskname")
+	}
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{azureconstants.RoleLabel: "Replica", azureconstants.VolumeNameLabel: diskname}}
```

Comment:
You can skip creating a label selector and directly create a map fill to `ListOptions` with `LabelSelector: labels.Set(map[string]string).String()`